src/WebAudio/Player.js

```diff
@@ -117,15 +154,60 @@ export default class Player extends AudioContext {
     }
 
     this.audioSource.stop();
+    this.pausedAt = 0;
+    this.startedAt = 0;
+    this.playing = false;
+
     this.fireEvent('stopped');
   }
 
+  /**
+   * Pause playback of audio.
+   */
   pause() {
     if (!this.audioBufferExists() && !this.audioSourceExists()) {
       return;
     }
 
+    const elapsed = this.audioContext.currentTime - this.startedAt;
     this.audioSource.stop();
+    this.pausedAt = elapsed;
```

Comment:
perhaps explain this a bit more. i'd say: pausedAt = this.audioContext.currentTime

Reply:
I'm not sure what you mean?
this.audioContext.currentTime holds the current time of the audioContext. It is not related to the audio playback, that is how AudioContext works.

In other words, pausing the playback doesn't pause the AudioContext.

Source: https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext/currentTime

Reply:
"It is not related to the audio playback": it is of course related, but you will have to calculate it against the moment you started audio playback.